Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -80,18 +100,40 @@ TEST_F(SerializerTest, EncodesNullProtoToBytes) {
   // sanity check (the _init_default above should set this to _NULL_VALUE)
   EXPECT_EQ(google_protobuf_NullValue_NULL_VALUE, proto.value.null_value);
 
-  /* NB: proto bytes were created via:
-       echo 'null_value: NULL_VALUE' \
-         | ./build/external/protobuf/src/protobuf-build/src/protoc \
-             -I./Firestore/Protos/protos \
-             -I./build/external/protobuf/src/protobuf/src/ \
-             --encode=google.firestore.v1beta1.Value \
-             google/firestore/v1beta1/document.proto \
-             > output.bin
-   */
+  // TEXT_FORMAT_PROTO: 'null_value: NULL_VALUE'
   std::vector<uint8_t> bytes{0x58, 0x00};
   ExpectRoundTrip(proto, bytes, FieldValue::Type::Null);
 }
 
+TEST_F(SerializerTest, EncodesBoolModelToProto) {
+  for (bool test : {true, false}) {
+    FieldValue model = FieldValue::BooleanValue(test);
+    Serializer::TypedValue proto{FieldValue::Type::Boolean,
+                                 google_firestore_v1beta1_Value_init_default};
+    proto.value.boolean_value = test;
+    ExpectRoundTrip(model, proto, FieldValue::Type::Boolean);
+  }
+}
+
+TEST_F(SerializerTest, EncodesBoolProtoToBytes) {
+  struct TestCase {
+    bool value;
+    std::vector<uint8_t> bytes;
+  };
+
+  // TEXT_FORMAT_PROTO: 'boolean_value: true'
+  std::vector<uint8_t> true_bytes{0x08, 0x01};
+  // TEXT_FORMAT_PROTO: 'boolean_value: false'
+  std::vector<uint8_t> false_bytes{0x08, 0x00};
+
+  for (const TestCase& test :
+       {TestCase{true, true_bytes}, TestCase{false, false_bytes}}) {
+    Serializer::TypedValue proto{FieldValue::Type::Boolean,
+                                 google_firestore_v1beta1_Value_init_default};
+    proto.value.boolean_value = test.value;
+    ExpectRoundTrip(proto, test.bytes, FieldValue::Type::Boolean);
+  }
+}
+
 // TODO(rsgowman): Test [en|de]coding multiple protos into the same output
```

Comment:
optional: death-test for decode invalid bytes? say, if there is data-corruption, I guess those should be caught by decoder.

Reply:
Yeah, we'll definitely need that. Error handling's not there yet though, so I've just added a TODO for now.